### DISCUSSION TITLE
Why arrows menu navigation is gone in 5.3.0?

### DISCUSSION THREAD

**Discussion by Solarunit** [tomasklaen/uosc#963] [Q&A] [answered]
Why arrows menu navigation is gone in 5.3.0?

**Comment by hooke007** [comment]
What does the `gone` mean?
up and down is over there. left and right is used to switch labels.

**Reply by Solarunit** [reply]
I mean left/right buttons to go to sub-menus

**Comment by tomasklaen** [comment, marked as answer]
Left and right arrows now navigate between main and side item actions (buttons). This is also possible with tab & shift+tab.

Submenu navigation is on enter and backspace.

I'm still not 100% sure which way is better. I thought that since there's currently only one menu in uosc that has submenus, the arrows would be more useful navigating actions, which are in almost all menus.

**Reply by Solarunit** [reply]
I think arrows are more obvious and faster for menu with submenus. For me submenu navigation on enter and backspace is ok, because I hardly even use main menu, I mostly use Subtitles and Audio menus directly.

**Reply by hooke007** [reply]
> I'm still not 100% sure which way is better.

If someone would implement the slider in the menu https://github.com/tomasklaen/uosc/issues/616 . Obviously `left` and `right` will not be preserved for navigation too.

**Comment by mrfragger** [comment]
yea I quite prefer the left/right arrows for menu navigation as it's intuitive.  Recently got someone to install uosc and they're a computer programmer and they couldn't navigate the menu till I said you need to hit ENTER.  I think tab should suffice along with the mouse for the side actions buttons.  When I navigate to open up subtitles it's much slower hitting ENTER 3 times.

**Reply by tomasklaen** [reply]
Well, it's hard to decide since only people who mind the current behavior post in this thread. But ok, I'll revert left/right to menu navigation.

Lets see how many threads with people complaining they can't select actions with arrows we'll get 😬.

**Reply by po5** [reply]
@tomasklaen Shouldn't right arrow also have the ability to enable menu items? Enter can be used for both navigation and activation, but right arrow can't.

**Reply by tomasklaen** [reply]
I decided against it. I don't like overloading arrows like that. They are just navigating now.

**Reply by hooke007** [reply]
Apparently, people could not reach the same view.
The final decision is up to whose voice is louder. LOL

**Comment by redomCL** [comment]
> Well, it's hard to decide since only people who mind the current behavior post in this thread. But ok, I'll revert left/right to menu navigation.
> 
> Lets see how many threads with people complaining they can't select actions with arrows we'll get 😬.



for my mind ,i'd like arrows left right to control!  and i like it on the outside so that the file name doesn't be override，on 5.2.0 lol......
![QQ截图20240904235900](https://github.com/user-attachments/assets/9b458cb1-56e4-4cc7-8008-2cdab68d2daf)

**Comment by Solarunit** [comment]
I have a couple of alternative ideas:

1. Long Enter press on item to open context menu like in Kodi
2. instead of buttons - submenu with text items (but we need some non-distracting indicator, that submenu is available)

**Reply by redomCL** [reply]
> I have a couple of alternative ideas:
> 
>     1. Long Enter press on item to open context menu like in Kodi
> 
>     2. instead of buttons - submenu with text items (but we need some non-distracting indicator, that submenu is available)

貌相更好，然后左右按键可以控制前进和后退
It's maybe better, and then the left and right buttons can control forward and backward like 5.2.0

**Reply by Solarunit** [reply]
if those buttons become sub-menus maybe we can use arrows for it :thinking: 
So question is can we use sub-menus for buttons?
